db/products.js

```diff
@@ -0,0 +1,69 @@
+let products = [];
+let validIDs = [];
+let lastID = 0;
+
+function all() {
+  return products;
+}
+
+function createID() {
+  let newID = lastID + 1;
+  lastID = newID;
+  validIDs.push(newID);
+  return newID;
+}
+
+function checkID(num) {
+  if (validIDs.indexOf(num) === -1) {
+    return false;
+  }
+  return true;
+}
+
+function findProductByID(inputID) {
+  let product = {};
+  for(let i=0; i<products.length; i++){
+    let current = products[i];
+    if(current.id === Number(inputID)){
+      product = current
+    }
+  }
+  return product;
+}
+
+function addProduct(responseObject) {
+  let newProduct = {};
+  newProduct.id = createID();
+  newProduct.name = responseObject.name;
+  newProduct.price = responseObject.price;
+  newProduct.inventory = responseObject.inventory;
+
+  products.push(newProduct);
+  return;
+}
+
+function editProduct(id, responseObject) {
+  let productEdit = findProductByID(id);
```

Comment:
Just for personal preference i would name this `product` since that's more holistically what it is and edit is the behavior you will take on it as per the function name.